packages/react-instantsearch-dom/src/hooks/useAnswers.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect, useMemo, useContext } from 'react';
+import { instantSearchContext } from 'react-instantsearch-core';
+import { createConcurrentSafePromise } from '../lib/createConcurrentSafePromise';
+import { debounce, debounceAsync } from '../lib/debounce';
+
+function hasReactHooks() {
+  // >= 16.8.0
+  const [major, minor] = React.version.split('.').map(Number);
+  if (major >= 17) {
+    return true;
+  }
+  if (major === 16 && minor >= 8) {
+    return true;
+  }
```

Comment:
```suggestion
  return Boolean(major >= 17 || (major === 16 && minor >= 8));
```

Not sure if that's better but it's a cute one-liner